src/crimson/osd/main_config_bootstrap_helpers.cc

```diff
@@ -143,6 +191,22 @@ _get_early_config(int argc, const char *argv[])
 	  std::begin(early_args),
 	  std::end(early_args));
 
+        std::optional<OptionValue> option_value;
+        for (const auto& option : seastar_options) {
+          option_value = get_option_value(option);
+          if (option_value) {
+            ret.early_args.emplace_back(option.option_name);
+            std::visit([&ret](auto&& value) {
+              using T = std::decay_t<decltype(value)>;
+              if constexpr (std::is_same_v<T, first_double> || std::is_same_v<T, second_double>) {
+                ret.early_args.emplace_back(std::to_string(value.value));
+              } else if constexpr (std::is_same_v<T, uint64_t>) {
+                ret.early_args.emplace_back(std::to_string(value));
+             }
+            }, *option_value);  // Dereference optional before using std::visit
+          }
+        }
```

Comment:
Going back to my previous comment, the loop could look something like this:
```
        std::optional<string> option_value;
        for (const auto& option : seastar_options) {
          option_value = get_option_value(option);
          if (option_value) {
            ret.early_args.emplace_back(option.option_name);
            ret.early_args.emplace_back(value);
           }
         }
```

What do you think?

Reply:
IIUC that is possible only when get_option_value return string type so if you don;t want i have to move this conversion code at get_option_value function but we have to convert it because all data type is not same.